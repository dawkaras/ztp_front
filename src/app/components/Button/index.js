import React from 'react';
import { string, func } from 'prop-types';
import Button from 'react-bootstrap/Button';

const ButtonComp = ({ text, onClick }) => (
  <Button
    color="primary"
    type="submit"
    onClick={() => onClick()}
    style={{ margin: '5px 0px' }}
  >
    {text}
  </Button>
);

ButtonComp.propTypes = {
  text: string.isRequired,
  onClick: func,
};

ButtonComp.defaultProps = {
  onClick: () => {},
};

export default ButtonComp;
