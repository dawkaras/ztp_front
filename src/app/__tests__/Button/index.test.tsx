import * as React from 'react';
import renderer from 'react-test-renderer';
import Button from 'app/components/Button';

it('renders Button component', () => {
  const component = renderer.create(<Button text="Test1" />).toJSON();
  expect(component).toMatchSnapshot();
});
it('renders Button component', () => {
  const component = renderer.create(<Button text="Test2" />).toJSON();
  expect(component).toMatchSnapshot();
});
