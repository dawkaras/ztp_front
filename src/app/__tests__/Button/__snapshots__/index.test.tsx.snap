// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component 1`] = `
<button
  className="btn btn-primary"
  color="primary"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "margin": "5px 0px",
    }
  }
  type="submit"
>
  Test1
</button>
`;

exports[`renders Button component 2`] = `
<button
  className="btn btn-primary"
  color="primary"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "margin": "5px 0px",
    }
  }
  type="submit"
>
  Test2
</button>
`;
